lib/foodcritic/xml.rb

```diff
@@ -23,7 +23,12 @@ def xml_hash_node(doc, xml_node, child)
       child.each do |c|
         n = xml_create_node(doc, c)
         c.drop(1).each do |a|
-          xml_node.add_child(build_xml(a, doc, n))
+          if a.first == :@label
+            label_node = xml_create_node(doc, a)
+            xml_node.add_child(build_xml(a, doc, label_node))
+          else
+            xml_node.add_child(build_xml(a, doc, n))
+          end
```

Comment:
can you explain what this is doing?  could probably use both a comment here, and a somewhat less terse explanation in the PR.  seems like this might be intrusive to other rules as well since it hits how we construct the XML representation of the AST.